dogma.go

```diff
@@ -385,20 +385,68 @@ func (c *Client) Push(ctx context.Context, projectName, repoName, baseRevision s
 	return c.content.push(ctx, projectName, repoName, baseRevision, commitMessage, changes)
 }
 
-// WatchFile awaits and returns the query result of the specified file since the specified last known revision.
-func (c *Client) WatchFile(ctx context.Context, projectName, repoName, lastKnownRevision string,
-	query *Query, timeout time.Duration) <-chan *WatchResult {
-	return c.watch.watchFile(ctx, projectName, repoName, lastKnownRevision, query, timeout)
+func (c *Client) watchWithWatcher(w *Watcher) (result <-chan WatchResult, closer func()) {
+	// setup watching channel
+	ch := make(chan WatchResult, DefaultChannelBuffer)
+	result = ch
+	w.Watch(func(value WatchResult) {
+		ch <- value
+	})
+
+	// setup closer
+	closer = func() {
+		w.Close()
+	}
+
+	// start watching
+	w.start()
+	return
+}
+
+// WatchFile watches on file changes. The watched result will be returned
+// through the returned channel. The api also provides manual closer to stop watching
+// and release underlying resources.
+// In short, watching will be stopped in case either context is cancelled or closer is
+// called.
+// Manually closing returned channel is unsafe and may cause sending on closed channel error.
```

Comment:
- Could you a small example snippet here?
- I'm curious if we can detect immediately when a user closes a channel. If so, we wouldn't need to provide a manual closer at all.

Reply:
In golang, if closes channel suddenly, then writing to it will result in a panic `writing to closed channel`. 
User should aware of it (just like `etcd` watching api) and not closing it explicitly.
In this way, I think we could provide snippet as you said and also warning user not closing channel himself but via `closer`